pkg/cmd/pulumi/new.go

```diff
@@ -1112,3 +1117,29 @@ func containsWhiteSpace(value string) bool {
 	}
 	return false
 }
+
+func validateStackProjectName(b backend.Backend, stackName, projectName string) error {
+	if stackName == "" {
+		return nil
+	}
+	if projectName == "" {
+		return nil
+	}
+
+	if strings.Count(stackName, "/") == 2 {
+		ref, err := b.ParseStackReference(stackName)
+		if err != nil {
+			return err
+		}
+		stackProjectName, hasProject := ref.Project()
+		if !hasProject {
+			return nil
+		}
+		if projectName == stackProjectName.String() {
+			return nil
+		}
+		return fmt.Errorf("project name (%s) and stack reference project name (%s) must be the same",
```

Comment:
Should we reference the flags here in the error message instead of the concept names?

e.g., something like:

    project name (--project %s) must match fully qualified stack reference (--stack %s)